spec/index.bs

```diff
@@ -3878,6 +3853,56 @@ Issue: link to the semantics of SV_SampleIndex and SV_Coverage in WGSL spec.
+
+            - [$validating GPUProgrammableStageDescriptor$]({{GPUShaderStage/VERTEX}},
+                |descriptor|.{{GPURenderPipelineDescriptor/vertexStage}},
+                |descriptor|.{{GPUPipelineDescriptorBase/layout}}) succeeds.
+            - If |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}} is not `null`:
+                - [$validating GPUProgrammableStageDescriptor$]({{GPUShaderStage/FRAGMENT}},
+                    |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}},
+                    |descriptor|.{{GPUPipelineDescriptorBase/layout}}) succeeds.
+            - If |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}} is `null`:
+                - |descriptor|.{{GPURenderPipelineDescriptor/colorStates}} is empty.
+            - |descriptor|.{{GPURenderPipelineDescriptor/colorStates}}.length is less than
+                or equal to 4.
+            - Each |colorState| layout descriptor in the list
+                |descriptor|.{{GPURenderPipelineDescriptor/colorStates}}:
+                - [$validating GPUColorStateDescriptor$](|colorState|,
+                    |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}}) succeeds.
+            - [$validating GPURasterizationStateDescriptor$](|descriptor|.{{GPURenderPipelineDescriptor/rasterizationState}},
+                |features|) succeeds.
+            - if |descriptor|.{{GPURenderPipelineDescriptor/depthStencilState}} is not `null`:
+                - [$validating GPUDepthStencilStateDescriptor$](|descriptor|.{{GPURenderPipelineDescriptor/depthStencilState}}) succeeds.
+            - [$validating GPUVertexStateDescriptor$](|descriptor|.{{GPURenderPipelineDescriptor/vertexState}},
+                |descriptor|.{{GPURenderPipelineDescriptor/vertexStage}}) succeeds.
+            - If |descriptor|.{{GPURenderPipelineDescriptor/alphaToCoverageEnabled}} is `true`:
+                - |descriptor|.{{GPURenderPipelineDescriptor/sampleCount}} is greater than 1.
+            - If the output SV_Coverage semantics is [=statically used=] by
+                |descriptor|.{{GPURenderPipelineDescriptor/fragmentStage}}:
+                - |descriptor|.{{GPURenderPipelineDescriptor/alphaToCoverageEnabled}} is `false`.
+            - If |descriptor|.{{GPURenderPipelineDescriptor/primitiveTopology}} is:
+                <dl class="switch">
+                    : {{GPUPrimitiveTopology/"line-strip"}} or
+                        {{GPUPrimitiveTopology/"triangle-strip"}}
+                    :: |descriptor|.{{GPURenderPipelineDescriptor/vertexState}}.{{GPUVertexStateDescriptor/indexFormat}}
+                        is not `undefined`
+                    : Otherwise
+                    :: |descriptor|.{{GPURenderPipelineDescriptor/vertexState}}.{{GPUVertexStateDescriptor/indexFormat}}
+                        is `undefined`
+                </dl>
+</div>
+
+Issue: define what "compatible" means for render target formats.
```

Comment:
Also we need to validate the matching between the FS and the VS

Reply:
Yes, added an issue